solidity/test/FullyBackedECDSAKeepFactoryTest.js

```diff
@@ -0,0 +1,1685 @@
+
+      await expectRevert(
+        keepFactory.isOperatorUpToDate(members[0], application),
+        "Operator not registered for the application"
+      )
+    })
+  })
+
+  describe("updateOperatorStatus", async () => {
+    let minimumBondableValue
+
+    before(async () => {
+      await initializeNewFactory()
+      await initializeMemberCandidates()
+      await registerMemberCandidates()
+
+      const pool = await FullyBackedSortitionPool.at(signerPool)
+      minimumBondableValue = await pool.getMinimumBondableValue()
+
+      await setUnbondedValue(members[0], minimumBondableValue.muln(3))
+    })
+
+    it("revers if operator is up to date", async () => {
+      await expectRevert(
+        keepFactory.updateOperatorStatus(members[0], application),
+        "Operator already up to date"
+      )
+    })
+
+    it("removes operator if bonding value has decreased below minimum", async () => {
+      currentValue = await bonding.unbondedValue(members[0])
+
+      const valueToWithdraw = currentValue.sub(minimumBondableValue).addn(1)
+
+      bonding.withdraw(valueToWithdraw, members[0], {from: members[0]})
+      assert.isFalse(
+        await keepFactory.isOperatorUpToDate(members[0], application),
+        "unexpected status of the operator after bonding value change"
+      )
+
+      await keepFactory.updateOperatorStatus(members[0], application)
+
+      await expectRevert(
+        keepFactory.isOperatorUpToDate(members[0], application),
+        "Operator not registered for the application"
+      )
+    })
+
+    it("does not update operator if bonding value increased insignificantly above minimum", async () => {
+      bonding.deposit(members[0], {value: new BN(1)})
+      assert.isTrue(
+        await keepFactory.isOperatorUpToDate(members[0], application),
+        "unexpected status of the operator after bonding value change"
+      )
+
+      await expectRevert(
+        keepFactory.updateOperatorStatus(members[0], application),
+        "Operator already up to date"
+      )
+    })
+
+    it("updates operator if bonding value increased significantly above minimum", async () => {
+      bonding.deposit(members[0], {value: minimumBondableValue.muln(2)})
+
+      assert.isFalse(
+        await keepFactory.isOperatorUpToDate(members[0], application),
+        "unexpected status of the operator after stake change"
+      )
+
+      await keepFactory.updateOperatorStatus(members[0], application)
+
+      assert.isTrue(
+        await keepFactory.isOperatorUpToDate(members[0], application),
+        "unexpected status of the operator after status update"
+      )
+    })
+
+    it("updates operator if bonding value decreased insignificantly above minimum", async () => {
+      bonding.withdraw(new BN(1), members[0], {from: members[0]})
+
+      assert.isFalse(
+        await keepFactory.isOperatorUpToDate(members[0], application),
+        "unexpected status of the operator after stake change"
+      )
```

Comment:
Should it be `isFalse` or `isTrue`? If it dropped insignificantly, the operator should still be up to date.

To be clear: I am fine with not having all `*insignificantly*` scenarios in `FullyBackedECDSAKeepFactoryTest.js` as I do agree it kinda duplicates sortition pool tests. What I want to have here all other paths added in this commit 👌 

We can leave it, we can drop it, I am fine with both.

Reply:
> Should it be `isFalse` or `isTrue`? If it dropped insignificantly, the operator should still be up to date.

https://github.com/keep-network/sortition-pools/blob/678a7b7d10274329fe77b36fe59a2f34e7fc0f65/contracts/FullyBackedSortitionPool.sol#L193-L195

It depends on how much it dropped. Due to the `floor` calculation for division an insignificant change has different implication on results between addition and subtraction. In this case an insignificant drop in unbonded value causes drop of the eligible weight so the change is noticable.